### ISSUE TITLE
Heated Steering Wheel turns on every day and charge limit keeps getting set to 100%

### ISSUE THREAD

**Issue by Schmee323** [alandtse/tesla#223]
## Version of the custom_component

v2.3.0

## Configuration

```yaml
2022-06-03 13:10:16 WARNING (MainThread) [homeassistant.components.media_player] Updating samsungtv_smart media_player took longer than the scheduled update interval 0:00:15
```

## Describe the bug

When i get in the car, the heated steering wheel turns on without me telling it to. It only does this when the custom integration is running. I have checked it against temperatures outside and it still does it on warm days regardless. When i disable the integration, it does not turn the heated steering wheel on by itself so seems to be linked. Also i have issues with the integration setting my charge limit to 100%. Sometimes i get in the car and it is fully charged without me changing the charge limit from either my phone or from the integration. I usually leave my charge limit at 85% but when the integration is running it will randomly set to 100%.

## Debug log

```text

Add your logs here.

```

**Comment by alandtse** [comment]
The component does not do anything command-wise unless you set it. You should review your logbook to see what is being triggered and investigate. Your HA may have an automation or your HA installation may be compromised.

**Comment by Schmee323** [comment]
Hi. I have checked that I have no automations or anything to do with the integration other that some history graphs. I have noticed that the steering wheel heater is turning on when I get in the car to leave the house and also when the car arrives home. As soon as I disable the integration, this bevahiour stops. I have no links to the integration other than displaying charge rate etc.

Simeon Larrea
0412818676
________________________________

**Comment by alandtse** [comment]
@Megabytemb @ehendrix23 any ideas about this one?

**Comment by Megabytemb** [comment]
I tend to agree with @alandtse's comments about how the integration (particularly the heated steering wheel and charge limit settings) does not have any logic that would automatically turn on those components.

I would encourage you to look in the home assistant Logbook, and see what fires in Home Assistant. It might not even be related to the car, just when you notice it. The Logbook lets you filter by entity, so you can see what (if anything) is turning on the heated steering wheel switch

You may have an automation that is accidentally turning on All switches, for example.

**Comment by magnevan** [comment]
This was happening to me too. I figured out that it was because I was using Node-Red and had set the "Device" option to my Tesla in the call service node. I thought it worked like a filter, but I guess it sends the command to every switch.

![Screenshot 2022-06-17 081155](https://user-images.githubusercontent.com/1482385/174326756-cbf82eb9-81af-4aa3-873e-3e2bdd1426e8.png)

**Comment by vadius45** [comment]
Thank you so much @magnevan : I was getting crazy with that erratic behavior

**Comment by Schmee323** [comment]
I hear you both, but I have never set up any automations linked to this integration, nor can I find any after extensive searching.

 

Here is a snapshot of the config and some of the logs from when I got home tonight at 6:41pm.

 

Looks like the max range switch is also turning on by itself which would explain the 100% charge target I mentioned. Is turning on at the same time.

 

For now I have disabled the integration as I cant have the charge level being reset every time I leave or arrive home. The heated steering wheel is an inconvenience but the charge limit has actual impact on battery health.

 

 

Happy to look further if you have any suggestions of where or how to debug this more.

 

Thanks,

Simeon
